---
title: "Using clusterProfiler to characterise Multi-Omics Data"
subtitle: "online example data"
author: "Maintained by YuLab-SMU"
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
  pdf_document:
    toc: true
---



```{r echo=FALSE, results="hide", message=FALSE, warning=FALSE}
library(fs)
library(knitr)
```
# Data description

## Inflammatory Bowel Disease example data description

NIHMS1510763-supplement-Dataset_1.xlsx: The Metabolomic Abundance Raw Matrix was downloaded from https://doi.org/10.1038/s41564-018-0306-4.

NIHMS1510763-supplement-Dataset_2.xlsx: The Metabolite Original ID and Metabolite Name Conversion Table was downloaded from https://doi.org/10.1038/s41564-018-0306-4.

NIHMS1510763-supplement-Dataset_6.xlsx: The Metagenomic Original Gene Abundance Matrix File was downloaded from https://doi.org/10.1038/s41564-018-0306-4.

enzyme_ko.txt: The ID mapping file (mapping enzymes to KEGG K-Numbers) was downloaded from https://github.com/YuLab-SMU/Supplemental-MMINP/blob/main/data/20220924_kegg/enzyme_ko.txt.

metabolwasm_expr.csv: The processed metabolomic abundance matrix file was derived from https://doi.org/10.1038/s41564-018-0306-4.

metabolwasm_meta.csv:  The processed metabolomic experimental grouping information file was derived from https://doi.org/10.1038/s41564-018-0306-4.

mg.expr.csv: The processed metagenomic abundance matrix file was derived from https://doi.org/10.1038/s41564-018-0306-4.

mg.meta.csv:  The processed metabolomic experimental grouping information file was derived from https://doi.org/10.1038/s41564-018-0306-4.

name_map.csv: The ID mapping file(mapping metabolites to KEGG C-Numbers) was downloaded from https://www.metaboanalyst.ca/MetaboAnalyst/resources/users/guest7299843510837833804tmp/name_map.csv.

## Phyllostachys Edulis example data description:

Phe_GO_annotation.txt: The Phyllostachys Edulis gene GO annotation data was downloaded from https://plantregmap.gao-lab.org/download_ftp.php?filepath=08-download/Phyllostachys_heterocycla/GO/Phe_GO_annotation.gz.

Phe_TF_list.txt: The Phyllostachys Edulis transcription factor family annotation data was downloaded from https://planttfdb.gao-lab.org/download/TF_list/Phe_TF_list.txt.gz.

regulation_from_motif_CE_Phe.txt: The Predicted Phyllostachys Edulis transcription factor target gene was downloaded from https://plantregmap.gao-lab.org/download_ftp.php?filepath=08-download/Phyllostachys_heterocycla/binding/regulation_from_motif_CE_Phe.txt.

counts.txt: The counts.txt was processed Phyllostachys Edulis gene count matrix derived from https://db.cngb.org/search/project/CNP0002243/.

group_info.txt: The Phyllostachys Edulis sample group infomation was derived from https://onlinelibrary.wiley.com/doi/10.1111/jipb.13357.

## PBMC single cell example data description:

c8.all.v2023.1.Hs.symbols.gmt: The C8 gene set was downloaded from https://www.gsea-msigdb.org/gsea/msigdb.

barcodes.tsv, genes.tsv, matrix.mtx: The single-cell transcriptomic data used in this study was derived from PBMC (Peripheral Blood Mononuclear Cell) samples and can be obtained from the following source: https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz.

# Data link
```{r echo=FALSE}
library(fs)
x1 <- dir_info('./')[,seq(3)] |>
      dplyr::filter(path=='examples') 
x <- dir_info('examples', recurse = TRUE)[, 1:3] |>
    as.data.frame() |>
    dplyr::filter(!grepl("Rmd$|html$|sh$|annot_data$|filtered_gene_bc_matrices$|hg19$|cell_marker_db$", path))

x <- rbind(x1, x)

x$path <- sprintf("[%s](%s)", x$path, x$path)

knitr::kable(x, row.names = FALSE)
```
